models.py

```diff
@@ -277,33 +297,43 @@ def _check_dims_and_sizes(self,
 
     def _compute_emissions(self,
                            words: Var,
+                           mask: Optional[Var] = None,
                            prefixes: Optional[Var] = None,
                            suffixes: Optional[Var] = None,
                            ) -> Var:
         assert words.dim() == 2
         batch_size, seq_length = words.size()
+        assert mask is None or mask.size() == (batch_size, seq_length)
         assert prefixes is None or prefixes.size() == (batch_size, seq_length, 2)
         assert suffixes is None or suffixes.size() == (batch_size, seq_length, 2)
 
-        # shape: (batch_size, seq_length + window)
-        padded = torch.cat((self._get_padding(batch_size), words), 1)  # pad left
-        # shape: (batch_size, seq_length + 2*window)
-        padded = torch.cat((padded, self._get_padding(batch_size)), 1)  # pad right
+        if self.window > 0:
+            # shape: (batch_size, seq_length + window)
+            words = torch.cat((self._get_padding(batch_size), words), 1)  # pad left
+            # shape: (batch_size, seq_length + 2*window)
+            words = torch.cat((words, self._get_padding(batch_size)), 1)  # pad right
+        
         # shape: (batch_size, seq_length + 2*window, word_embedding_size)
-        embedded_words = self._embed_words(padded)
+        embedded_words = self._embed_words(words)
         # shape: (batch_size, seq_length, total_prefix_embedding_size)
         embedded_prefixes = self._embed_prefixes(prefixes)
         # shape: (batch_size, seq_length, total_suffix_embedding_size)
         embedded_suffixes = self._embed_suffixes(suffixes)
 
-        temp = []
-        for i in range(seq_length):
-            lo, hi = i, i + 2 * self.window + 1
-            # shape: (batch_size, (2*window + 1) * word_embedding_size)
-            word_feats = embedded_words[:, lo:hi, :].contiguous().view(batch_size, -1)
-            temp.append(word_feats)
-        # shape: (batch_size, seq_length, (2*window + 1) * word_embedding_size)
-        word_features = torch.stack(temp, dim=1)
+        embedded_words = self.dropout(embedded_words)
+        embedded_prefixes = self.dropout(embedded_prefixes)
+        embedded_suffixes = self.dropout(embedded_suffixes)
```

Comment:
Line 324 and 325 will raise errors if `use_prefix=False` or `use_suffix=False` respectively. This is because `embedded_prefixes/suffixes` will be `None` when the corresponding flag is `False`. Could you handle this?